src/Datadog.Trace/DuckTyping/DuckType.cs

```diff
@@ -666,6 +666,16 @@ private T ThrowOnError<T>(object instance)
         /// <typeparam name="T">Type of proxy definition</typeparam>
         public static class CreateCache<T>
         {
+            /// <summary>
+            /// Gets the type of T
+            /// </summary>
+            public static readonly Type Type = typeof(T);
```

Comment:
Out of curiosity, did this field in particular bring some improvement? I don't know what's the cost of `typeof(T)`

Reply:
`typeof(T)` is translated to a `ldtoken` of T and then a call to `System.Type.GetTypeFromHandle` which is an [extern call](https://source.dot.net/#System.Private.CoreLib/src/System/Type.CoreCLR.cs,95).  The improvement is not so big, but because I use this twice in the `IsVisibility` and in the `#if NET45` from `GetProxySlow` it looks like a good opportunity to cache it. I have plans to do a bigger refactor to improve a bit more the `Generics` execution path in the future, and this could be helpful as well.